interface/main/calendar/add_edit_event.php

```diff
@@ -1582,14 +1731,32 @@ function find_available(extra) {
     dateChanged();
 });
 
+function are_days_checked(){
+    var days = document.getElementById("days").getElementsByTagName('input');
+    var counter = 0;
+    for(var i=0; i < days.length; i++){
+       if(days[i].checked){
+           counter++;
+       }
+    }
+    return counter;
+}
+
 // Check for errors when the form is submitted.
 function validate(valu) {
      var f = document.getElementById('theform');
-    if (f.form_repeat.checked &&
+    if ((f.form_repeat.checked || f.days_every_week.checked) &&
         (! f.form_enddate.value || f.form_enddate.value < f.form_date.value)) {
         alert('<?php echo addslashes(xl("An end date later than the start date is required for repeated events!")); ?>');
         return false;
     }
+
+    //Make sure if days_every_week is checked that at least one weekday is checked.
+    if(f.days_every_week.checked && !are_days_checked()){
+        alert('<?php echo addslashes(xl("Must choose at least one day!")); ?>');
```

Comment:
here's a clearer short-cut function for this (xls combines xl and addslashes):
``````
alert('<?php echo xls("Must choose at least one day!"); ?>');